plugins/woocommerce/src/Internal/EmailEditor/BlockEmailRenderer.php

```diff
@@ -112,4 +126,34 @@ private function render_block_email( \WP_Post $email_post, string $woo_content,
 	private function get_email_post_by_wc_email( \WC_Email $email ): ?\WP_Post {
 		return $this->template_manager->get_email_post( $email->id );
 	}
+
+	/**
+	 * Build email context from WC_Email object.
+	 *
+	 * Extracts relevant context data from the WC_Email object that can be used
+	 * by blocks during rendering, such as user ID, email address, order information, etc.
+	 *
+	 * Blocks that need cart product information can derive it from the user_id or email
+	 * using CartCheckoutUtils::get_cart_product_ids_for_user().
+	 *
+	 * @param \WC_Email $wc_email WooCommerce email object.
+	 * @return array Email context data.
+	 */
+	private function build_email_context( \WC_Email $wc_email ): array {
+		$recipient_raw = $wc_email->get_recipient();
+		$emails        = array_values( array_filter( array_map( 'sanitize_email', array_map( 'trim', explode( ',', $recipient_raw ) ) ) ) );
+		$context       = array(
+			'recipient_email' => $emails[0] ?? null,
+			'woo_email_type'  => $wc_email->id,
+		);
+
+		// Extract order-related context if the email object is an order.
+		if ( isset( $wc_email->object ) && $wc_email->object instanceof \WC_Order ) {
+			$order               = $wc_email->object;
+			$context['order_id'] = $order->get_id();
+			$context['user_id']  = $order->get_customer_id();
+		}
+
+		return $context;
+	}
```

Comment:
I remember seeing a similar piece of code and I found it [here](https://github.com/woocommerce/woocommerce/blob/8b1f23220364e796d172c79dbf0ed98a69369520/plugins/woocommerce/src/Internal/EmailEditor/TransactionalEmailPersonalizer.php#L67-L81).
It looks like now we have two separate contexts (for rendering and for personalization tags). Do you think we could unify that and use one context for both?